napari/layers/points/points.py

```diff
@@ -279,18 +405,100 @@ def data(self, data: np.ndarray):
+                    edge_color_property_value = self.current_properties[
+                        self._edge_color_property
+                    ][0]
+
+                    ec, _ = map_property(
+                        prop=edge_color_property_value,
+                        colormap=self.edge_colormap[1],
+                        contrast_limits=self._edge_contrast_limits,
+                    )
+                    new_edge_colors = np.tile(ec, (adding, 1))
                 self.edge_color = np.vstack((self.edge_color, new_edge_colors))
-                new_face_colors = np.tile(
-                    self._current_face_color, (adding, 1)
-                )
+
+                # add new face colors
+                if self._face_color_mode == ColorMode.DIRECT:
+                    new_face_colors = np.tile(
+                        self._current_face_color, (adding, 1)
+                    )
+                elif self._face_color_mode == ColorMode.CYCLE:
+                    face_color_property_value = self.current_properties[
+                        self._face_color_property
+                    ][0]
+                    new_face_colors = np.tile(
+                        self.face_color_cycle_map[face_color_property_value],
+                        (adding, 1),
+                    )
+                elif self._face_color_mode == ColorMode.COLORMAP:
+                    face_color_property_value = self.current_properties[
+                        self._face_color_property
+                    ][0]
+
+                    fc, _ = map_property(
+                        prop=face_color_property_value,
+                        colormap=self.face_colormap[1],
+                        contrast_limits=self._face_contrast_limits,
+                    )
+                    new_face_colors = np.tile(fc, (adding, 1))
                 self.face_color = np.vstack((self.face_color, new_face_colors))
+
                 self.size = np.concatenate((self._size, size), axis=0)
+
         self._update_dims()
         self.events.data()
 
+    @property
+    def properties(self):
+        """ dict {str: array (N,)}, DataFrame: Annotations for each point"""
+        return self._properties
+
+    @properties.setter
+    def properties(self, properties: Dict[str, np.ndarray]):
+        if not isinstance(properties, dict):
+            properties = dataframe_to_properties(properties)
+        self._properties = self._validate_properties(properties)
+        if self._face_color_property and (
+            self._face_color_property not in self._properties
+        ):
+            self._face_color_property = ''
+            warnings.warn('property used for face_color dropped')
+
+    def _validate_properties(self, properties: Dict[str, np.ndarray]):
+        """Validates the type and size of the properties"""
+        for v in properties.values():
+            if not isinstance(v, np.ndarray):
+                raise TypeError('all properties should numpy arrays')
```

Comment:
I think this check is too stringent. If someone wants to give us an ndarray like object here (e.g. a jax array), as long as everything works as expected, we should be ok with that. (`pandas.Series` is one relevant example).

Reply:
That makes sense to me. I will remove the type check and leave the length check.